crypto/rand/randfile.c

```diff
@@ -109,10 +109,15 @@ int RAND_load_file(const char *file, long bytes)
     }
 
     if (bytes < 0) {
-        if (S_ISREG(sb.st_mode))
+        if (S_ISREG(sb.st_mode)) {
+#if SIZE_MAX > ULONG_MAX
```

Comment:
@bernd-edlinger just to be sure: is this the correct and portable way to compare `sizeof(size_t)` and `sizeof(long)` in the preprocessor?

Reply:
effectively we don't know if st_size is off_t aka. long or off64_t aka long long,
so there is neither a size_t nor a unsigned long involved.

Reply:
you could for instance silence the warning here, this way:
`bytes = (sb.st_size < LONG_MAX) ? sb.st_size : LONG_MAX;`

Reply:
Yes, that seems to be the proper way to check the size difference.

Reply:
*However*, to be safe, you should include `"internal/numbers.h"

Reply:
> I think I never understood, why this size <= LONG_MAX was done in the first place.

I don't understand anymore either why I did it that way. I thought I had done it to fix a compiler warning.  But when I retry with

https://github.com/mspncp/openssl/blob/85eb60a5aa3f93516cd1974b3b7259542e341594/crypto/rand/randfile.c#L111-L116

I don't get any warnings on Linux 64bit. Also the VC64 compiler does not complain. Maybe I mixed it up with a size_t warning further down, I don't recall.

```
[00:05:24] ..\crypto\rand\randfile.c(144): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
```
https://ci.appveyor.com/project/mspncp/openssl/builds/19861056/job/q5x102tvsglqp44w

I'll just remove the conditional expression entirely and do a normal assignment.